code/modules/integrated_electronics/subtypes/input.dm

```diff
@@ -686,6 +730,49 @@
 	spawn_flags = IC_SPAWN_DEFAULT|IC_SPAWN_RESEARCH
 	power_draw_per_use = 120
 
+/obj/item/integrated_circuit/input/sensor/ranged/sense_remote(var/atom/A,mob/user)
+	if(!user)
+		if(user.client)
```

Comment:
there's no user, how can you check client of a non-existent user?

Reply:
there is user.It is called with afterattack.

Reply:
`if(!user)`